src/components/TPS6208x.stanza

```diff
@@ -120,3 +120,102 @@ public pcb-module device (pn:TPS6208x-PN -- snooze-mode:True|False, snooze-conn?
       val [p1, p2] = get-element-ports(pull)
       net (SW.MODE, p1)
       net (p2, snooze-ep)
+
+doc: \<DOC>
+Construct a Buck Converter Circuit for TPS6208x series
+
+@member conv
+@member enable
+@member power-good
+<DOC>
+public defn circuit (
+  pn:TPS6208x-PN,
+  cxt:BuckConstraints,
+  buck-sol?:Maybe<BuckSolution> = One(MFR-Suggested-Solution()),
+  --
+  snooze-mode:True|False,
+  snooze-conn?:Instantiable = ?
+  C-query?:CapacitorQuery = ?
+  ) -> Instantiable:
+
+  pcb-module circuit-type:
+    port conv : power-converter
+    port enable
+    port power-good
+
+    public inst IC : device(pn, snooze-mode = snooze-mode, snooze-conn? = snooze-conn?)
+
+    net GND (conv.VIN.V-, conv.VOUT.V-, IC.buck.gnd)
+    net (conv.VIN.V+, IC.buck.vin)
+
+    val L-series = None()
+    val C-series = E12()
+    val C-query = match(C-query?):
+      (_:None): CapacitorQuery(
+          type = "ceramic",
+          temperature-coefficient_code = "X7R"
+          case = ["0603", "0805", "1206"]
+          precision = precision(C-series)
+        )
+      (given:One<CapacitorQuery>): value(given)
+
+
+    val sol = match(buck-sol?):
+      (_:None): compute-solution(cxt)
+      (given:One<BuckSolution>): value(given)
+
+    val vout-max = max-value(v-out(cxt))
+    val vin-max = max-value(v-in(cxt))
+
+    ; TODO - Check vin max against the input voltage spec
+    ;   for the chip.
+
+    val V-margin = cap-V-margin(cxt)
+    val C-in-type = create-capacitor(
+      C-query
+      capacitance = C-in(sol),
+      rated-voltage = AtLeast(vin-max * V-margin)
```

Comment:
Is this search looking for a particular rated-voltage or is it like a min search (i.e. accept anything with this value or greater)?

Reply:
No - it is a minimum voltage rating. It could select for greater. I'm expecting that the user is going to use the `C-query` input to make more decisions about how to select parts. For example - by using `sort! = SortKey(rated-voltage = Increasing)` or something.